### ISSUE TITLE
[Accessibility & Usability] Home Screen Colors Causing Loss of Usability

### ISSUE THREAD

**Issue by glenn-sorrentino** [guardianproject/haven#228]
The use of the background video/image and white text & icons makes the app very hard to use because of low contrast. 

![default screen issues](https://user-images.githubusercontent.com/28545431/35120039-af71326c-fc4b-11e7-972f-dab318aed64a.jpg)

Issues numbered in the slide below:

1. Using white on low contrast backgrounds makes the text above the
timer and the timer itself nearly impossible to read.

2. Hard to read the button text, and the visual treatment is inconsistent with more space above the text than on the left and right sides. This is also inconsistent with the button style in the tour.

3. When I started this audit, I didn’t see this button. It’s 6 hours later and I just discovered it.

6. The icons are visible, but only because there happens to be something with high contrast on the wall.

Side-By-Side:
![home-side-by-side](https://user-images.githubusercontent.com/28545431/35121001-cea4c0b0-fc4e-11e7-8975-2c176d07a69e.png)

Current Home Screen:
![home - current](https://user-images.githubusercontent.com/28545431/35121014-da208ae6-fc4e-11e7-925e-05382df04f0b.png)

Proposed Design:
![home - proposed](https://user-images.githubusercontent.com/28545431/35120869-61f09b88-fc4e-11e7-9d1f-e9c5ada579be.png)

**Comment by fat-tire** [comment]
Does everyone agree with this?  I can do it but it does involve removing buttons so I want to get everyone's agreement.

Potential issue:  The placement of the top buttons (left, centered, and then right) in the top menu bar thing are not standard for Android.  What is behind the left overflow buttons?  And what does the middle "haven" button do?  Usually the menu items are on the right by default, but just wanted to get some clarification.

Also, @glenn-sorrentino do you have a clean replacement "blue" background image to use?

**Comment by fat-tire** [comment]
Also what is that white central camera icon with the right-facing arrow connected to?   I do like the design- it's clean and appealing.  Just not sure what everything does...

**Comment by glenn-sorrentino** [comment]
@fat-tire For the placement of the Settings there are two main goals:

- Create consistent placement of the icons across screens in the app
   - On the Logs screen, the icon is in the top right corner, on the default screen, it's on the bottom right. 

- Use the Material Design guidelines for navigation & settings
   - For example, Google uses the 3 vertical dots to store this kind of content. You can see an example [in the Material Design guidelines](https://material.io/guidelines/patterns/settings.html#).

Since the app doesn't have a lot of navigation, I'm just showing the settings icon so that the user can save a tap, but you make a good point that the pattern for Settings is usually inside of a menu, not necessarily shown by default. 

I've iterated on two new designs with your feedback. One uses the dot menu, the other uses a consistent treatment for the Settings icon, similar to the Logs page.

[Haven Settings & Navigation.pdf](https://github.com/guardianproject/haven/files/1648508/Haven.Settings.Navigation.pdf)

![haven settings navigation - cover](https://user-images.githubusercontent.com/28545431/35179650-4c4e2466-fd53-11e7-932a-4f99dafc9f79.jpg)
![haven settings navigation 1](https://user-images.githubusercontent.com/28545431/35179651-5036c0d8-fd53-11e7-8953-ee7d130e1a9d.jpg)
![haven settings navigation 2](https://user-images.githubusercontent.com/28545431/35179654-558fae00-fd53-11e7-8104-3b25e2bc25c7.jpg)
![haven settings navigation 3](https://user-images.githubusercontent.com/28545431/35179657-58f492a4-fd53-11e7-9c26-7a12a9f44fbc.jpg)
![haven settings navigation 4](https://user-images.githubusercontent.com/28545431/35179659-5be7d7be-fd53-11e7-8973-4605b23c2f5f.jpg)
![haven settings navigation 5](https://user-images.githubusercontent.com/28545431/35179660-5d9b8b50-fd53-11e7-8bbd-11e75f08d786.jpg)

**Comment by glenn-sorrentino** [comment]
[Material guidelines](https://material.io/guidelines/layout/structure.html) also shows a title in each screen. Instead of a literal text title, the Haven icon serves this purpose. It's non-actionable and reinforces the brand.

**Comment by glenn-sorrentino** [comment]
@fat-tire Lastly, the camera icon changes which camera Haven will use. This is from the [icon set](https://material.io/icons/#ic_camera_rear) in Material Design. I don't think that there's any change needed on your end.

**Comment by glenn-sorrentino** [comment]
@fat-tire For the clean background image, I don't think it should be a still image necessarily - just a layer between the text and the video to make the text accessible. There may be a few ways of doing this, but my thought was to programmatically use a background color and give it an 80% opacity. The color I'm using is rgba(20, 94, 169, .8).

We'd also need to include a drop shadow on any text. This ensures that when there is a background color that might swing the contrast back to being too low (the light on the ceiling may cause this), it gives enough contrast between the letterform and whatever color is behind it.

If you need any assets from me - a transparent PNG or anything else, please let me know!

**Comment by fat-tire** [comment]
Hey nice work as before.  I started implementing this but came across some issues.  Here are some thoughts/observations:

*  Ah, I see that white camera icon is the front/back camera toggler thing.

*  I am using a transparent dark blue background as you suggest-- with all the yellow at first it doesn't look awesome, but anyway it settles down and looks better.

* The menu bar title in the material guidelines (and in every app I've seen) is usually some kind of text.  I think if it's gonna be the centered Haven shield, maybe it should be consistent across the whole app?

* When the "Start Now" button is hit, the clock timer starts to count down.  The "Show Logs" button transforms to a "Deactivate" button.  When the timer reaches zero, there is a large DEACTIVATE button. 
 Also, the time counter is itself a button that when tapped allows the user to set the countdown timer.

I added the semi-transparent background, removed the buttons at the bottom, added the overflow menu on the right, but when I started pulling buttons that had actual functionality (like the countdown), I started to wonder if that's what we wanted to do..

So I'm stopping to get some feedback on what you think the functionality might look like.  I do like the simplicity of the fewer buttons, but what do we do with the counter?  Or where do we put it?

**Comment by glenn-sorrentino** [comment]
Good point about the title - I agree about a consistent treatment - I'm not married to the icon -
 maybe it should be text instead since the user can navigate to Logs or Settings, and each screen should have a unique title.

### Start Now
For the "Start Now" interaction - I think that the "Deactivate" button can appear after the timer has been started. An analogous use case is when you start a timer on the native clock app - after you start it, the "Delete" and "Add Timer" options are progressively displayed.

### View Logs
My hypothesis is that by moving the Logs button to the menu would help to decrease poential user error - for example, if I see the "View Logs" button, then choose to press "Start Now" - I might accidentally press "Deactivate" because the button was just "View Logs". It could be a frustrating lesson to learn. 

### Demo
I created a quick demo of what I think the interaction could look like. It explores color to convey state,  introduces a timer when Haven is active and simplifies its presentation. I think that this would be valuable feedback to the user - when they pick their device up, they have immediate confirmation that it's been running for 2 hours, for example.

Check it out:

![haven-timer](https://user-images.githubusercontent.com/28545431/35182117-614a865e-fd83-11e7-872b-22475c68e198.gif)

**Comment by n8fr8** [comment]
I've updated the code in the app, so that now the camera icon opens the camera configuration screen, which allows both camera switching and sensitivity control.

The idea of having all the sensor icons on that screen, and direct access to the sensitivity control, is that it will be a very frequent, common action to tune each sensor for the current environment. Having it buried under two clicks in settings, makes that a bit harder to reach.

**Comment by n8fr8** [comment]
I guess that means I am responding to this comment
"Granular settings that a user won’t use often shouldn’t be so prominent in the UI."
to say, these aren't granular, and they will be used often.

**Comment by glenn-sorrentino** [comment]
Ah, makes total sense. What do you think putting those options in the settings menu?

**Comment by glenn-sorrentino** [comment]
![settings-menu](https://user-images.githubusercontent.com/28545431/35187133-f012bb22-fdd3-11e7-84aa-ca006c81bb43.png)

**Comment by glenn-sorrentino** [comment]
Since this issue is about the background color/accessibility, maybe we should move this conversation to the issue addressing the UI? https://github.com/guardianproject/haven/issues/229

**Comment by fat-tire** [comment]
I had started on adjusting the UI, but as it's still under discussion I'll hold off until there's consensus.  I'll meet you all at #229 :)

**Comment by n8fr8** [comment]
I am all for the background color layer over the video feed. It looks great, and addresses the usability and accessibility issues.
